label.py

```diff
@@ -359,8 +360,34 @@ def show_help():
     print('======================================================================================\n')
 
 
+def images_cmp(x, y):
+    img_list_x = x.split('_')
+    img_list_y = y.split('_')
+    if len(img_list_x) != len(img_list_y):
+        return len(img_list_x) - len(img_list_y)
+    for (f1, f2) in zip(img_list_x, img_list_y):
+        if f1 == f2:
+            continue
+        if f1.isdigit():
+            return int(f1) - int(f2)
+
+
+def group_images():
+    sorted_images_to_show = sorted(images_to_show, key=functools.cmp_to_key(images_cmp))
+    bug_ids = set([file_name.split('_')[0] for file_name in sorted_images_to_show])
+    ordered_images = []
+    for bug_id in bug_ids:
+        for file_name in sorted_images_to_show:
+            if bug_id == file_name.split('_')[0]:
+                ordered_images.append(file_name)
```

Comment:
Better to use a list comprehension here, similar to what you were doing before, but with two for loops in the comprehension rather than one.

Reply:
ordered_images = [XXX for bug_id in bug_ids for file_name in sorted_images_to_show if bug_id == ...]